graphx/src/main/scala/org/apache/spark/graphx/lib/StronglyConnectedComponents.scala

```diff
@@ -106,6 +116,11 @@ object StronglyConnectedComponents {
           }
         },
         (final1, final2) => final1 || final2)
+
+        // ensure sccGraph's rdd are marked as recently used to not be evicted between iterations
```

Comment:
OK, the rest of the changes much looks like an improvement, at the least. I am not as sure why this is necessary. It's already materialized at this point.